### PR TITLE
Make resource migration more efficient

### PR DESCRIPTION
**Pull request by tanya-borisova**
# Resolves https://github.com/microsoft/AzureTRE/issues/2253

## What is being addressed

Migrations that add deploymentStatus to resources are taking a long time on main (because resources on main have a lot of operations associated with them).

In this PR, making it more efficient, via only querying resources that do not already have a deploymentStatus field.

### CODE REVIEW

**Review by damoodamoo — APPROVED** [review]

File: templates/workspaces/base/porter.yaml [hunk 1]
```diff
@@ -96,7 +96,7 @@ parameters:
   - name: app_service_plan_sku
     type: string
     description: "The SKU used when deploying an Azure App Service Plan"
-    default: "P1v3"
+    default: "P1v2"
```
Comment (ross-p-smith) [inline templates/workspaces/base/porter.yaml:99]: I thought that this would need to be a default of P1v3 - as that is what we want customers to use

Comment (tanya-borisova) [reply templates/workspaces/base/porter.yaml:99]: Uh oh, this slipped in from my testing! Thanks for the catch

**Pull request merged**